### DISCUSSION TITLE
Dynamic Workflows

### DISCUSSION THREAD

**Discussion by migoe101** [Intermesh/groupoffice#866] [General]
It would be nice, if it was possible to set dynamic next Steps in Workflows.

For Exampel: Approval of payable Invoices

1. Folder "Incoming Invoices" of all incoming Invoices set to Workflow Invoice Approval
2. Step 1 - Save Invoice in Folder "Incoming Invoices"
2. Step 2 - Accountant proof the adress, VAT and so on after that approve and move file to: proof by ...  - Select next Step: 3 (1. CEO, 2. Production, 3. Logistics)
3. Step 3 - Logisitcs approve invoice and move to: to book
4. Step 4 - Accountant book invoice, approv and move to: Invoice Archiv

**Comment by michalcharvat** [comment]
Not sure what do you have to do after step 2 -> go to Step 3? Or Step 3.1 and only when 3.1 is done you will move to the 3.2? Approve by one of these 3? Still possible via standard module options. 

You can always split steps into small pieces. 

Earlier I have made modification where was possible set different steps when you approved or declined current step but looks like you case is different.

**Comment by migoe101** [comment]
Yes I mean something else. In my example, what is meant is that the approver in step 2 can determine whether the approver in step 3 should be either 3.1 CEO, 3.2 Production or 3.3 Logistics.

Currently, this would require several smaller processes, which of course have to be restarted accordingly.
